test/tests/val-2f/val-2f.i

```diff
@@ -15,32 +15,37 @@ desorption_duration = '${fparse (temperature_desorption_max-temperature_desorpti
 endtime = '${fparse charge_time + cooldown_duration + desorption_duration}'
 
 # Materials properties
-diffusion_W_preexponential = '${units 1.6e-7 m^2/s}'
+diffusion_W_preexponential = '${units 1.6e-7 m^2/s -> mum^2/s}'
 diffusion_W_energy = '${units 0.28 eV -> J}'
 
 # Source term parameters
-sigma = '${units 0.5e-9 m}'
-R_p = '${units 0.7e-9 m}'
-fluence = '${units 1.5e25 atoms/m^2}'
-flux = '${units ${fparse fluence / charge_time} atoms/m^2/s}'
+sigma = '${units 0.5e-9 m -> mum}'
+R_p = '${units 0.7e-9 m -> mum}'
+fluence = '${units 1.5e25 at/m^2 -> at/mum^2}'
+flux = '${units ${fparse fluence / charge_time} at/m^2/s -> at/mum^2/s}'
 
 # Numerical parameters
 dt_max_large = '${units 100 s}'
 dt_max_small = '${units 10 s}'
-dt_start_charging = '${units 1 s}'
-dt_start_cooldown = '${units 10 s}'
-dt_start_desorption = '${units 1 s}'
-
-# Geometry and mesh
-length_W = '${units 0.8 mm}'
-num_nodes_W = 40
+# dt_start_charging = '${units 1 s}'
+# dt_start_cooldown = '${units 10 s}'
+# dt_start_desorption = '${units 1 s}'
 
 [Mesh]
-  [generated]
-    type = GeneratedMeshGenerator
+  [cartesian_mesh]
+    type = CartesianMeshGenerator
     dim = 1
-    nx = ${num_nodes_W}
-    xmax = ${length_W}
+    dx = '${units 1e-6 m -> mum}
```

Comment:
As I said, you need the mesh to have roughly 5 to 20 elements across the implantation section. The implantation section length scale is much smaller than that. 
Also, make sure the total length of the elements amount to the desired sample size.